src/Components/Main.tsx

```diff
@@ -86,145 +69,194 @@ export const Main = (props: IProps) => {
     return comparison;
   };
 
-  const filteredRequests = requests.slice(0, props.maxRequests).filter((request) => {
-    if (filter === EnumFilterType.All) {
-      if (request instanceof ReduxAction) {
-        return request.action.type.toLowerCase().includes(searchQuery.toLowerCase());
+  const filteredRequests = (): Array<any> =>
+    requests.slice(0, props.maxRequests).filter(request => {
+      if (filter === EnumFilterType.All) {
+        if (request instanceof ReduxAction) {
+          return JSON.stringify(request.action).toLowerCase().includes(searchQuery.toLowerCase());
+        }
       }
-    }
-    // TODO: Add  || request instanceof NRequest when Native request will be available
-    if (request instanceof RNRequest) {
-      if (filter === request.method || filter === EnumFilterType.All) {
-        return (
-          request.url?.toLowerCase().includes(searchQuery.toLowerCase()) ||
-          request.status === parseInt(searchQuery) ||
-          request.method.toLowerCase() === searchQuery.toLowerCase()
-        );
+      if (request instanceof RNRequest || request instanceof NRequest) {
+        if (request.type === EnumSourceType.ReactNativeRequest) {
+          if (filter === request.method || filter === EnumFilterType.All) {
+            return (
+              request.type === EnumSourceType.ReactNativeRequest &&
+              (request.url?.toLowerCase().includes(searchQuery.toLowerCase()) ||
+                request.status === parseInt(searchQuery, 10) ||
+                request.method.toLowerCase() === searchQuery.toLowerCase())
+            );
+          }
+        } else {
+          if (filter === request.method || filter === EnumFilterType.All) {
```

Comment:
what is the difference with `EnumSourceType.ReactNativeRequest` case to split it? Look like the same code

Reply:
If we choose to display only the NativeRequest, search must continue to work. That's why I have a duplicate here. However, after I'll do tests unit on this part, I'll try to do a refactor on this part.

Reply:
Done